build/package/proxy.Dockerfile

```diff
@@ -0,0 +1,25 @@
+# Build the manager binary
+FROM golang:1.13.1 as builder
+
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# Copy the go source
+COPY cmd/ cmd/
+COPY api/ api/
+COPY internal/ internal/
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o proxy cmd/proxy/main.go
```

Comment:
`-a` isn't required - it forces rebuilds of packages but go already knows when to rebuild things.

Reply:
The container is blank before this so it's going to need to build it all anyway. Regardless, I'd like to err on the side of caution and make it force build anyway?

Reply:
🤷‍♂ I'm not too fussed, but I'd recommend *not* enabling it personally... I don't think there's really anything to protect against. It used to be more relevant in the earlier days of Go. Additionally, it'd be great in future to utilise something like https://github.com/kubernetes-sigs/kind/blob/master/hack/go_container.sh which allows efficient runs of `go` that preserve the cache, to make it much faster 😄